tests/unit/base/trained_model_serialization_unittest.cc.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+
+def find_guard(content):
+    guards = ['USE_CPLEX', 'USE_GPL_SHOGUN', 'HAVE_LAPACK', 'USE_SWIG_DIRECTORS']
```

Comment:
these guards should not be hard coded, but rather come from the cmake setup. 
In fact, this whole thing should not be done in a python script I think
@vigsterkr will have comments on how to organise this